File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/ItemsDoesNotLookInApplicatorsValidCaseTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class ItemsDoesNotLookInApplicatorsValidCaseTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(new JsonSchemaKeywordFlags.Builder().format().build());

    @Test
    public void testPrefixitemsInAllofDoesNotConstrainItemsValidCasePasses() throws ValidationException {
        // prefixItems in allOf does not constrain items, valid case
        final var schema = ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCase1.getInstance();
        schema.validate(
            new ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCaseListBuilder()
                .add(5)

                .add(5)

            .build(),
            configuration
        );
    }

    @Test
    public void testPrefixitemsInAllofDoesNotConstrainItemsInvalidCaseFails() {
        // prefixItems in allOf does not constrain items, invalid case
        final var schema = ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCase1.getInstance();
        try {
            schema.validate(
                Arrays.asList(
                    3,
                    5
                ),
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/ItemsDoesNotLookInApplicatorsValidCaseTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class ItemsDoesNotLookInApplicatorsValidCaseTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testPrefixitemsInAllofDoesNotConstrainItemsValidCasePasses() {
        // prefixItems in allOf does not constrain items, valid case
        val schema = ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCase1.getInstance()
        schema.validate(
            ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCaseListBuilder()
                .add(5)

                .add(5)

            .build(),
            configuration
        )
    }

    @Test
    fun testPrefixitemsInAllofDoesNotConstrainItemsInvalidCaseFails() {
        // prefixItems in allOf does not constrain items, invalid case
        val schema = ItemsDoesNotLookInApplicatorsValidCase.ItemsDoesNotLookInApplicatorsValidCase1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    listOf(
                        3,
                        5
                    ),
                    configuration
                )
            }
        )
    }
}
